### PR TITLE
Story #4035 Desktop backorder refactoring bug

### PR DESCRIPTION
**Pull request by jrbirch99**
- Add a context flag to skip refactoring

### CODE REVIEW

**Review by armandcela — APPROVED** [review]

**Review by PeterAlabaster — APPROVED** [review]

File: addons/udes_stock/models/stock_move.py [hunk 1]
```diff
@@ -687,3 +689,16 @@ def _refactor_action_by_maximum_quantity(self, maximum_qty):
                 new_pickings += picking._backorder_movelines(mls, dest_picking)
 
         return self | new_pickings.mapped("move_lines")
+
+    def _split(self, qty, restrict_partner_id=False):
+        """
+        If the moves have movelines we do not want to refactor the moves when we split them, as
```
Comment (PeterAlabaster) [inline addons/udes_stock/models/stock_move.py:695]:
```suggestion
        Extend _split to disable move refactoring when the move has move_lines linked to it
        ..then something here as to why, i don't follow what we mean by `as the movelines will not be refactored`
```

File: addons/udes_stock/models/stock_move.py [hunk 2]
```diff
@@ -687,3 +687,14 @@ def _refactor_action_by_maximum_quantity(self, maximum_qty):
                 new_pickings += picking._backorder_movelines(mls, dest_picking)
 
         return self | new_pickings.mapped("move_lines")
+    
+    def _split(self, qty, restrict_partner_id=False):
+        """
+        If the moves have movelines we do not want to refactor the moves when we split them, as
+        the movelines will not be refactored.
+        """
+        if self.mapped("move_line_ids"):
```
Comment (armandcela) [inline addons/udes_stock/models/stock_move.py:698]: It looks like self is always 1 record as even in core odoo is treated like that self.product_uom_qty etc.
I think don't need the mapped in this case,

File: addons/udes_stock/models/stock_move.py (hunk 1 above)
Comment (jrbirch99) [reply addons/udes_stock/models/stock_move.py]: changed :slightly_smiling_face:

File: addons/udes_stock/models/stock_move.py (hunk 2 above)
Comment (jrbirch99) [reply addons/udes_stock/models/stock_move.py]: changed :slightly_smiling_face:

**Pull request merged**